### ISSUE TITLE
RedisClient 3.8.5 - Full waiting queue issue

### ISSUE THREAD

**Issue by glassfox** [vert-x3/vertx-redis-client#216]
Hi Guys, 
Found scenario when RedisClient do not forward exception in case of full waiting queue.

Version: 3.8.5

Following the test:

 ```
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;
import java.util.stream.IntStream;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import io.vertx.core.CompositeFuture;
import io.vertx.core.Future;
import io.vertx.core.Promise;
import io.vertx.core.Vertx;
import io.vertx.core.net.SocketAddress;
import io.vertx.junit5.Timeout;
import io.vertx.junit5.VertxExtension;
import io.vertx.junit5.VertxTestContext;
import io.vertx.redis.client.Redis;
import io.vertx.redis.client.RedisAPI;
import io.vertx.redis.client.RedisOptions;
import io.vertx.redis.client.Response;

@ExtendWith(VertxExtension.class)
@Timeout(10000)
public class RedisClientThreadSafeTest {

	private RedisAPI redisApi;
	private Redis redis;
	
	@BeforeEach
	void setUp(Vertx vertx, VertxTestContext context) throws Exception {
		
		var options = new RedisOptions();
		options.addEndpoint(SocketAddress.inetSocketAddress(6379, "localhost")).setMaxWaitingHandlers(10);
		
		Redis.createClient(vertx, options)
		.connect(onConnect -> {
				if (onConnect.succeeded()) {
					redis = onConnect.result();
					redisApi = RedisAPI.api(redis);
					context.completeNow();
				} else {
					context.failNow(onConnect.cause());
				}
		});
	}

	@AfterEach
	void tearDown() throws Exception {
		redis.close();
	}

	@Test
	void test(Vertx vertx, VertxTestContext context) {
		AtomicInteger cnt = new AtomicInteger();
		List<Future> futures = new ArrayList<>();
		IntStream.range(0, 100).forEach(i ->{
			Promise<Response> p = Promise.promise();
			vertx.setTimer(1, timerid ->{
				redisApi.set(List.of("foo","bar"), p);
			});
			futures.add(p.future().map(res -> {
				System.out.println("SUCCESS " + cnt.incrementAndGet());
				return null;
			}));
		});
		
		CompositeFuture.all(futures).setHandler(res ->{
			if(res.failed()) {
				res.cause().printStackTrace();
				context.failNow(res.cause());
			} else {
				context.completeNow();
			}
		});
	}
}
```

**Comment by pmlopes** [comment]
Hi @glassfox could you point out what the issue is? 

When the queue is full you want the handler for the command to be notified with that error? or the generic exception handler?

**Comment by glassfox** [comment]
Hi @pmlopes ,
Response to the handler of the command - is the must have behavior.
In case of generic exception handle: I saw in the original code, comment about dynamic enchantments of the queue... For the management of the queue size maybe ??? Do you have any additional scenarios ?

**Comment by pmlopes** [comment]
Ok, I understand it better now. I'll work on it today.

**Comment by pmlopes** [comment]
the fix has been backported to 3.9.0
